_posts/2020-09-30-ghelamco-alert.md

```diff
@@ -0,0 +1,824 @@
+
+AWS greengrass offers solutions to all these challenges!  
+Greengrass is an extension of the AWS IoT service and since we had already set up our device in AWS IoT it was easy for us to set up greengrass.  
+To get started with greengrass we had to do 2 additional steps:
+* Define a greengrass group. This group will contain your IoT devices and deployments.  
+* Define a greengrass core. The core is the device that you will use to run the additional AWS capabilities on.  
+
+The greengrass group allows us to push deployments to all the devices in our group.  
+It also allows you to run lambda functions on your core device or install additional connectors with your own runtime of choice.  
+<div style="text-align: center;">
+  <img alt="Greengrass group" src="/img/2020-09-25-ghelamco-alert/gg_group.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+When you register your device as a core device in greengrass you immediately get some nice additional benefits from this.  
+For example, you can immediately see all the network interfaces on your core device and what ip addresses got allocated.  
+This is especially useful if you want to ssh to your device and do not have fixed ip attached to it.  
+<div style="text-align: center;">
+  <img alt="Greengrass core" src="/img/2020-09-25-ghelamco-alert/gg_core.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+Now we that greengrass is installed and our RPI is configured as a core device our greengrass group we can start making full use of the capabilities that greengrass offers.  
+We decided that we wanted our application to run as a docker container, so we installed and configured the **docker connector** for greengrass.   
+This plugin allows you to run docker containers on your core device and makes use of **docker** and **docker compose**.  
+<div style="text-align: center;">
+  <img alt="Greengrass docker connector" src="/img/2020-09-25-ghelamco-alert/gg_connector_docker_cfg.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+For this to work you have to install docker and docker compose on your core device.  
+Since we are using a RPI it was a bit harder to install since the RPI uses the ARM7 chipset instead of x64/x32.  
+We had no trouble installing docker itself, but for docker compose we had to clone the source code from github and compile the binaries ourself on the RPI.  
+After all the setup was done we could just create a docker container in our CICD pipeline and tell greengrass to run that container on RPI.  
+
+### CICD pipeline RPI Backend
+
+To make sure we did not need to bother ourselves with manual builds and installs of our code on the RPI we built a CICD pipeline to automatically deploy our software onto the RPI.  
+We trigger the pipeline whenever a push to our master branch in our git repository.  
+We used azure devops as our CICD system.  
```

Comment:
link to azure devops

Note: why azure devops and doesn't aws have something similar?

Reply:
because bousson has an azure fetish